src/OpenTelemetry.Instrumentation.AspNet/Implementation/HttpInListener.cs

```diff
@@ -129,6 +130,15 @@ public override void OnStartActivity(Activity activity, object payload)
                 activity.SetTag(SemanticConventions.AttributeHttpUserAgent, request.UserAgent);
                 activity.SetTag(SemanticConventions.AttributeHttpUrl, request.Url.ToString());
 
+                if (this.options.RecordHttpServerAttributes)
+                {
+                    var xForwardedFor = request.ServerVariables["HTTP_X_FORWARDED_FOR"];
+                    if (!string.IsNullOrEmpty(xForwardedFor))
+                    {
+                        activity.SetTag(SemanticConventions.AttributeHttpClientIP, xForwardedFor.Split(',').First().Trim());
+                    }
```

Comment:
Should we add an `else` block that sends `context.Request.ServerVariables["REMOTE_ADDR"]` as ClientIP if `X-Forwarded-For` isn't available?

Reply:
Added